code/identity/src/main/java/com/adobe/marketing/mobile/identity/IdentityExtension.java

```diff
@@ -894,14 +869,14 @@ boolean handleSyncIdentifiers(
         }
 
         // if the marketingCloudServer is null or empty use the default server
-        if (StringUtils.isNullOrEmpty(currentEventValidConfig.marketingCloudServer)) {
-            currentEventValidConfig.marketingCloudServer = IdentityConstants.Defaults.SERVER;
+        if (StringUtils.isNullOrEmpty(latestValidConfig.marketingCloudServer)) {
+            latestValidConfig.marketingCloudServer = IdentityConstants.Defaults.SERVER;
```

Comment:
I find it odd that we update properties of latestValidConfig as part of this method. Instead ConfigurationSharedStateIdentity should take care of replacing null/empty values with the default server

Reply:
ConfigurationSharedStateIdentity already sets marketingCloudServer to the default if null/empty, so this check is not needed. I went ahead and made this class immutable as it's odd to have a class like this which is mutable.